crates/cargo-platform/src/cfg.rs

```diff
@@ -124,30 +145,40 @@ impl fmt::Display for Cfg {
         match *self {
             Cfg::Name(ref s) => s.fmt(f),
             Cfg::KeyPair(ref k, ref v) => write!(f, "{} = \"{}\"", k, v),
+            Cfg::Version(CfgRustVersion { minor, patch }) => {
+                write!(f, "version(\"1.{minor}.{patch}\")")
+            }
         }
     }
 }
 
 impl CfgExpr {
     /// Utility function to check if the key, "cfg(..)" matches the `target_cfg`
-    pub fn matches_key(key: &str, target_cfg: &[Cfg]) -> bool {
+    pub fn matches_key(key: &str, target_cfg: &[Cfg], rustc_version: &semver::Version) -> bool {
         if key.starts_with("cfg(") && key.ends_with(')') {
             let cfg = &key[4..key.len() - 1];
 
             CfgExpr::from_str(cfg)
                 .ok()
-                .map(|ce| ce.matches(target_cfg))
+                .map(|ce| ce.matches(target_cfg, rustc_version))
                 .unwrap_or(false)
         } else {
             false
         }
     }
 
-    pub fn matches(&self, cfg: &[Cfg]) -> bool {
+    pub fn matches(&self, cfg: &[Cfg], rustc_version: &semver::Version) -> bool {
         match *self {
-            CfgExpr::Not(ref e) => !e.matches(cfg),
-            CfgExpr::All(ref e) => e.iter().all(|e| e.matches(cfg)),
-            CfgExpr::Any(ref e) => e.iter().any(|e| e.matches(cfg)),
+            CfgExpr::Not(ref e) => !e.matches(cfg, rustc_version),
+            CfgExpr::All(ref e) => e.iter().all(|e| e.matches(cfg, rustc_version)),
+            CfgExpr::Any(ref e) => e.iter().any(|e| e.matches(cfg, rustc_version)),
+            CfgExpr::Value(Cfg::Version(CfgRustVersion { minor, patch })) => {
+                match minor.cmp(&rustc_version.minor) {
+                    Ordering::Less => true,
+                    Ordering::Equal => patch <= rustc_version.patch,
+                    Ordering::Greater => false,
+                }
+            }
```

Comment:
I believe the tracking issue mentions specific behavior for nightlies
- `1.80` includes 1.80 nightlies
- `1.80.0` does not include 1.80 nightlies

Reply:
I'm not quite following this. Can you say more about where that is being discussed? That doesn't seem to be the current behavior.

Reply:
Maybe I'm misinterpeting what was said in

> Ultimately, a compromise was agreed upon, in which nightly releases are treated as "complete" i.e. cfg(version(1.20)) evals to true on nightly builds with version 1.20.0, but there is a nightly flag -Z assume-incomplete-release to opt into the behaviour that doesn't do this assumption. This compromise was implemented in PR https://github.com/rust-lang/rust/pull/81468.

The actual rustc tests should be checked to confirm.  We probably should most or all of their tests.

Reply:
Did I get the desired logic here?

https://github.com/rust-lang/cargo/blob/120bea75161530b7e4d3c26c1d5da32c95d740e3/crates/cargo-platform/tests/test_cfg.rs#L224-L234

https://github.com/rust-lang/cargo/blob/120bea75161530b7e4d3c26c1d5da32c95d740e3/crates/cargo-platform/src/cfg.rs#L36-L52

Reply:
@epage Any update on that?

I will have time on Tuesday or Wednesday to finish the PR after clarifying the desired behavior for nightly.